Persephone/Controllers/QueueViewController.swift

```diff
@@ -81,25 +82,28 @@ class QueueViewController: NSViewController, NSOutlineViewDataSource, NSOutlineV
     guard let queue = notification.userInfo?[Notification.queueKey] as? [MPDClient.Song]
       else { return }
 
-    self.queue = queue
+    var newQueue: [SongItem] = []
 
+    for (index, mpdSong) in queue.enumerated() {
```

Comment:
If you like, this could be expressed fairly nicely with `map`:

```swift
self.queue = queue.enumerated().map { index, song in
  SongItem(song: song, queuePos: index, isPlaying: index == queuePos)
}
```